bot/exts/events/trivianight/_questions.py

```diff
@@ -0,0 +1,240 @@
+        """
+        if all("visited" in question.keys() for question in self.questions) and number is None:
+            return Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="All of the questions in the question bank have been used.",
+                color=Colours.soft_red
+            )
+
+        if number is None:
+            question_number = randrange(0, len(self.questions))
+            while "visited" in self.questions[question_number].keys():
+                question_number = randrange(0, len(self.questions))
+        else:
+            question_number = number - 1
+
+        self.questions[question_number]["visited"] = True
+
+        # The `self.view` refers to the QuestionView
+        self.view.current_question = self.questions[question_number]
+
+    def list_questions(self) -> Union[Embed, str]:
+        """
+        Lists all questions from the question bank.
+
+        It will put the following into a message:
+            - Question number
+            - Question description
+            - If the question was already 'visited' (displayed)
+        """
+        if not self.questions:
+            return Embed(
+                title=choice(NEGATIVE_REPLIES),
+                description="No questions are currently loaded in!",
+                color=Colours.soft_red
+            )
+        spaces = len(
+            sorted(
+                self.questions, key=lambda question: len(question['description'])
+            )[-1]["description"]
+        ) + 3
+        formatted_string = ""
+        for question in self.questions:
+            question_description = question["description"]
+            formatted_string += f"`Q{question['number']}: {question_description}" \
+                                f"{' ' * (spaces - len(question_description) + 2)}" \
+                                f"|` {':x:' if not question.get('visited') else ':white_check_mark:'}\n"
+
+        return formatted_string.strip()
+
+    def current_question(self) -> tuple[Embed, QuestionView]:
+        """Returns an embed entailing the current question as an embed with a view."""
+        return self.view.create_current_question(), self.view
+
+    def end_question(self) -> Embed:
+        """Terminates answering of the question and displays the correct answer."""
+        scores, answer_embed, time_limit, total_points = self.view.end_question()
+        for user, score in scores.items():
+            time_taken = score[2]
+            point_calculation = (1 - (time_taken / time_limit) / 2) * total_points
+            if score[-1] is True:
+                self.scoreboard[UserScore(user)] = {"points": point_calculation, "speed": time_taken}
+            elif score[-1] is False and score[2] <= 2:
+                # Get the negative of the point_calculation to deduct it
+                self.scoreboard[UserScore(user)] = {"points": -point_calculation}
```

Comment:
Wait now why exactly so we remove as much as the user would've won if it wasn't for *whatever the if-statement tests*?

Reply:
Since if the user responds in less than 2 seconds and gets it wrong, they would've lost the points they were going to gain.

I should honestly make the score into a dictionary so it's clearer or document each part of score.